lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7772,6 +7772,22 @@ static void finishInheritedConformances(
   }
 }
 
+static Type
+recursivelySubstituteBaseType(const NormalProtocolConformance *conformance,
+                              DependentMemberType *depMemTy) {
+  Type origBase = depMemTy->getBase();
+  if (auto *depBase = origBase->getAs<DependentMemberType>()) {
+    Type substBase = recursivelySubstituteBaseType(conformance, depBase);
+    ModuleDecl *module = conformance->getDeclContext()->getParentModule();
+    return depMemTy->substBaseType(module, substBase);
```

Comment:
This definitely needs a check from someone who understands the ramifications better than I do.

Reply:
This feels like it should be a `Type::subst()`, not its own subst-like recursive algorithm.

Reply:
Oops, I forgot this change was in here, or I wouldn't have merged until you got a chance to review. IIRC we can't use `Type::subst` because we're still in the process of building the conformance, but I could be wrong.